worlds/landstalker/__init__.py

```diff
@@ -0,0 +1,247 @@
+        unfilled_location_count = len(self.multiworld.get_unfilled_locations(self.player))
+        while len(item_pool) < unfilled_location_count:
+            item_pool.append(self.create_item("EkeEke"))
+
+        self.multiworld.itempool += item_pool
+
+    def create_teleportation_trees(self):
+        self.teleport_tree_pairs = load_teleport_trees()
+
+        # Shuffle teleport tree pairs if the matching setting is on
+        if self.multiworld.shuffle_trees[self.player].value == 1:
+            all_trees = []
+            for pair in self.teleport_tree_pairs:
+                all_trees += [pair[0], pair[1]]
+            self.multiworld.random.shuffle(all_trees)
+            self.teleport_tree_pairs = []
+            while len(all_trees) > 0:
+                self.teleport_tree_pairs.append([all_trees[0], all_trees[1]])
+                all_trees = all_trees[2:]
+
+        # If a specific setting is set, teleport trees are potentially active without visiting both sides.
+        # This means we need to add those as explorable paths for the generation algorithm.
+        teleport_trees_mode = self.multiworld.teleport_tree_requirements[self.player].value
+        created_entrances = []
+        if teleport_trees_mode < 2:
+            for pair in self.teleport_tree_pairs:
+                entrances = create_entrance(pair[0]['region'], pair[1]['region'], True, self.player, self.regions_table)
+                created_entrances += entrances
+        if teleport_trees_mode == 1:  # Teleport trees are open but require access to Tibor in order to work
+            for entrance in created_entrances:
+                entrance.access_rule = make_path_requirement_lambda(self.player, [], [self.regions_table['tibor']])
+
+    def set_rules(self):
+        Rules.create_rules(self.multiworld, self.player, self.regions_table, self.dark_region_ids)
+
+        # In "Reach Kazalt" goal, player doesn't have access to Kazalt, King Nole's Labyrinth & King Nole's Palace.
+        # As a consequence, all locations inside those regions must be excluded and the teleporter from
+        # King Nole's Cave to Kazalt must go to the end region instead.
+        if self.get_setting("goal").value == 1:
+            kazalt_tp = self.multiworld.get_entrance("king_nole_cave -> kazalt", self.player)
+            kazalt_tp.connected_region = self.regions_table["end"]
+
+            EXCLUDED_REGIONS = [
+                "kazalt",
+                "king_nole_labyrinth_pre_door",
+                "king_nole_labyrinth_post_door",
+                "king_nole_labyrinth_exterior",
+                "king_nole_labyrinth_fall_from_exterior",
+                "king_nole_labyrinth_raft_entrance",
+                "king_nole_labyrinth_raft",
+                "king_nole_labyrinth_sacred_tree",
+                "king_nole_labyrinth_path_to_palace",
+                "king_nole_palace"
+            ]
+
+            for location in self.multiworld.get_locations(self.player):
```

Comment:
In cases like this, I find it clearer to have the condition inside the for rather than using a list comprehension.
If the change is absolutely required, I can do it, but I'd prefer not to (but it's just a matter of style honestly)